src/parsing.js

```diff
@@ -102,10 +102,83 @@
     return matchedUnits;
   }
 
+  function ignore(value) {
+    return function(input) {
+      var result = value(input);
+      if (result)
+        result[0] = undefined;
+      return result;
+    }
+  }
+
+  // Regex should be anchored with /^
+  function consumeToken(regex, string) {
+    var result = regex.exec(string);
```

Comment:
This & the other consume methods were a bit confusing to me. I think it would be clearer if we renamed "result" to "matched" or similar here (but keep using "result" for the pairs below).